README.md

```diff
@@ -33,6 +33,13 @@ In modern web frameworks, composable fragments of an application have the follow
 All of these properties directly affect the quality of the user experience, in terms of performance and smoothness.
 To achieve this kind of user experience today, developers have to rely on JavaScript libraries or frameworks that manage in-document “components”, “pages”, “layout”, loading states etc.
 
+## Design principles
```

Comment:
These principles make sense to me. Rather than pursuing either low-level primitives or high-level declarative features, it's doing both. I think customizable select is a recent successful example of this, first shipping the building blocks like popover and anchor positioning, and finally customizable select.